src/SqlPersistence/Subscription/SubscriptionPersister.cs

```diff
@@ -66,6 +74,44 @@ public SubscriptionPersister(Func<DbConnection> connectionBuilder, string tableP
     {
         var types = messageTypes.ToList();
 
+        if (cacheFor == null)
+        {
+            return await GetSubscriptions(types).ConfigureAwait(false);
+        }
+
+        var typeNames = types.Select(_ => _.TypeName);
+        var key = string.Join(",", typeNames);
+        CacheItem cacheItem;
+        if (cache.TryGetValue(key, out cacheItem))
+        {
+            if (DateTimeOffset.UtcNow - cacheItem.Stored < cacheFor)
+            {
+                return cacheItem.Subscribers;
+            }
+        }
+
+        var baseSubscribers = await GetSubscriptions(types)
+            .ConfigureAwait(false);
+
+        cacheItem = new CacheItem
+        {
+            Stored = DateTimeOffset.UtcNow,
+            Subscribers = baseSubscribers.ToList()
+        };
+
+        cache.AddOrUpdate(key, s => cacheItem, (s, tuple) => cacheItem);
+
+        return cacheItem.Subscribers;
+    }
+
+    class CacheItem
+    {
+        public DateTimeOffset Stored;
```

Comment:
TBH i did not think so. but NH uses it. since we are in memory and always DTC i think change it to datetime?